### PR TITLE
Support new Debian version of gmock

### PR DESCRIPTION
**Pull request by jspricke**
Debian moved gmock to /usr/src/googletest/googlemock, according to the
upstream move into the googletest project. This patch adds explicit
search patches for the old and new environment.

Also, it searches for gmock.cc instead of CMakeLists.txt, as the later
is too generic.

### CODE REVIEW

**Comment by SirVer** [comment]
@jspricke This breaks building on trusty - see travis. Could you have another look?

**Comment by jspricke** [comment]
Looks like you don't have libgtest-dev intalled on trusty.

**Comment by damonkohler** [comment]
In that case, you should be able to modify the Dockerfiles to fix the Travis build.

**Comment by jspricke** [comment]
Looks like the gtest in google-mock is newer then the one in libgtest-dev. Will try to find a new solution later.

**Comment by SirVer** [comment]
@jspricke I rebased your work on top of master and added travis configuration so that debian is not regressing in the future. Could you please sign the CLA so that we can get this merged?

This fixes https://github.com/googlecartographer/cartographer/issues/177 as well.

**Comment by jspricke** [comment]
I think this should be signed already by Magazino, or do I need to do anything @SirVer?

**Comment by SirVer** [comment]
dunno. @damonkohler probably knows.

**Comment by jihoonl** [comment]
I have added @jspricke's email(github@jochen.sprickerhof.de) in Magazino CLA about a week ago. Does it only work with `magazino.eu` email?

**Comment by SirVer** [comment]
Probably. This is copied from an internal help text about company CLAs:

> As part of signing the CLA, the company designates a Google Group that identifies who at their company is authorized to contribute to Google projects. If someone is trying to contribute to your project, but they are not in their company's authorized contributor group, they'll need to contact whoever administers the group for their company.

**Comment by SirVer** [comment]
@jihoonl Actually, now rereading our internals maybe I misunderstood CLABot. Probably it was happy because it saw the correct email address already - I only looked into the database for individual contributor. 

So in fact this can be merged now. Sorry for the noise.

File: .travis.yml [hunk 1]
```diff
@@ -22,12 +22,13 @@ cache:
     - /home/travis/docker/
 
 env:
-  - UBUNTU_RELEASE=trusty DOCKER_CACHE_FILE=/home/travis/docker/trusty-cache.tar.gz
-  - UBUNTU_RELEASE=xenial DOCKER_CACHE_FILE=/home/travis/docker/xenial-cache.tar.gz
+  - LSB_RELEASE=trusty DOCKER_CACHE_FILE=/home/travis/docker/trusty-cache.tar.gz
+  - LSB_RELEASE=xenial DOCKER_CACHE_FILE=/home/travis/docker/xenial-cache.tar.gz
+  - LSB_RELEASE=stretch DOCKER_CACHE_FILE=/home/travis/docker/stretch-cache.tar.gz
```
Comment (wohe) [inline .travis.yml]: Should we also add jessie? Assuming that it works of course.

**Pull request merged**